src/future/cache.rs

```diff
@@ -830,9 +844,12 @@ where
         key: Arc<K>,
         hash: u64,
         init: impl Future<Output = V>,
+        replace_if: Option<impl Fn(&V) -> bool>,
```

Comment:
looks like this function is relied on by many methods of the cache, maybe the performance loss on adding a new `match` assertion should be benchmarked.

Reply:
Thank you for the suggestion. The `get_with` method will be affected by the `match` condition change although it should have relatively smaller performance impact than calling `self.base.get_with_hash(&key, hash)` (which already exists in older versions).

Just for sure, I created a micro benchmark using [Criterion](https://crates.io/crates/criterion) to compare performance of `get_with` method between previous Moka v0.8.3 and the current version. 

- https://gitlab.com/moka-labs/moka-gh-issue-123/moka-microbench/-/blob/main/benches/future_cache_benchmark.rs

I run it on my Mac mini M1 and Criterion reported "No change in performance detected."

```console
## Moka v0.8.3 (baseline)
$ cargo bench --features moka083
...

get_with                time:   [2.2915 us 2.2938 us 2.2964 us]                      
Found 5 outliers among 100 measurements (5.00%)
  2 (2.00%) high mild
  3 (3.00%) high severe

## Moka with current change
$ cargo bench --features moka084
...

get_with                time:   [2.3039 us 2.3064 us 2.3093 us]                      
                        change: [-0.1492% +0.2773% +0.6093%] (p = 0.16 > 0.05)
                        No change in performance detected.
Found 7 outliers among 100 measurements (7.00%)
  2 (2.00%) low mild
  2 (2.00%) high mild
  3 (3.00%) high severe
```